config/v1/types_apiserver.go

```diff
@@ -30,6 +30,15 @@ type APIServerSpec struct {
 	// - ConfigMap.Data["ca-bundle.crt"] - CA bundle.
 	// +optional
 	ClientCA ConfigMapNameReference `json:"clientCA"`
+	// extraCORSAllowedOrigins is a field of allowed CORS that should be added on top of the basic CORS needed by the
+	// infrastructure in order to allow in-cluster components to access the API and OAuth servers using Javascript.
+	// The field values should follow the syntax of the corsAllowedOrigins field.
+	// If any operator means to use the field, it should only be checking that the value that it requires to appear
+	// is/is not present, and add it if it isn’t.
+	// We can assume that the array of the values of the field is managed by other operators, so the operator managing
+	// a certain value should remove it in case the operator itself is being deleted. This removal must happen before
+	// the resource that is represented by the value being removed (e.g. hostname from a route status) is itself removed.
```

Comment:
It may be implied here already but any operator editing this field must also handle changes to its exposed URL (i.e. remove the old URL and add the new one).

Reply:
I added a sentence about changes to the values.